packages/ui-pattern-block/src/UIPatternBlock.tsx

```diff
@@ -1,25 +1,238 @@
+        hasRadius,
+        radiusChoice,
+        radiusValue,
+        description,
+        title,
+    } = { ...DEFAULT_BLOCK_SETTINGS, ...blockSettings };
+    const { debounce } = useDebounce();
+    const isEditing = useEditorState(appBridge);
+    const [isResponsivePreviewOpen, setIsResponsivePreviewOpen] = useState(false);
+    const [dependencies, setDependencies] = useState(blockDependencies);
+    const [reset, setReset] = useState(false);
+
+    const cssToInject = useMemo(() => {
+        const alignmentCss = alignment === Alignment.Center ? centeringCss : '';
+        const backgroundCss = hasBackground ? getBackgroundCss(backgroundColor) : '';
+        const hasAutoHeight = !isCustomHeight && heightChoice === Height.Auto;
+        return getScriptToInject(getCssToInject(hasAutoHeight, alignmentCss, backgroundCss));
+    }, [hasBackground, backgroundColor, alignment, isCustomHeight, heightChoice]);
+
+    const templateFiles = useMemo(
+        () => ({ ...getDefaultFilesOfTemplate(sandpackTemplate), ...files?.[sandpackTemplate] }),
+        [sandpackTemplate, cssToInject, dependencies, reset],
+    );
+
+    const onResetToDefault = () => {
+        setReset(!reset);
+    };
+
+    const onCodeChange = (filename: string, code: string) => {
+        if (!isEditing) {
+            return;
+        }
+        debounce(() =>
+            setBlockSettings({
+                files: {
+                    ...blockSettings.files,
+                    [sandpackTemplate]: {
+                        ...blockSettings.files?.[sandpackTemplate],
+                        [filename]: code,
+                    },
+                },
+            }),
+        );
+    };
+
+    const onDependenciesChanged = (newDependencies: string, source: 'npm' | 'external') => {
+        if (isEditing) {
+            setBlockSettings({
+                dependencies: {
+                    ...blockSettings.dependencies,
+                    [sandpackTemplate]: {
+                        ...blockSettings.dependencies?.[sandpackTemplate],
+                        [source]: newDependencies,
+                    },
+                },
+            });
+        }
+        setDependencies({
+            ...dependencies,
+            [sandpackTemplate]: {
+                ...dependencies?.[sandpackTemplate],
+                [source]: newDependencies,
+            },
+        });
+    };
+
+    const npmDependencies = dependencies?.[sandpackTemplate]?.npm ?? '';
+    const externalDependencies = dependencies?.[sandpackTemplate]?.external ?? '';
+
+    const parsedNpmDependencies = useMemo(() => {
+        try {
+            return { dependencies: JSON.parse(npmDependencies) };
+        } catch {
+            return {};
+        }
+    }, [dependencies]);
+
+    const parsedExternalDependencies = useMemo(() => {
+        try {
+            return JSON.parse(externalDependencies);
+        } catch {
+            return [];
+        }
+    }, [dependencies]);
```

Comment:
They are both almost the same, can we have them like that:

```
const useParsedDependencies = (dependenciesString: string) => {
    return useMemo(() => {
        try {
            return JSON.parse(dependenciesString);
        } catch {
            return {};
        }
    }, [dependenciesString]);
};
```

And then:

```
const parsedNpmDependencies = useParsedDependencies(npmDependencies);
const parsedExternalDependencies = useParsedDependencies(externalDependencies);
```

Reply:
Created a helper function for these.